eZ/Publish/SPI/Search/Indexer.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace eZ\Publish\SPI\Search;
+
+/**
+ * Indexer for handlers that index stuff.
+ */
+interface Indexer
```

Comment:
> Because it can result in deletion of Content.

That assumption will become wrong once we do stuff like 9b7ec0d47e9cb06b3cc4fcc3dff5f3515146fc86. With changes like that it is up to signals to know if a given location operation affects content or not, so more clear separation.